util/src/main/java/io/camunda/zeebe/util/sched/WorkStealingGroup.java

```diff
@@ -12,7 +12,7 @@
 import java.util.concurrent.ThreadLocalRandom;
 
 /** Workstealing group maintains a queue per thread. */
-public final class WorkStealingGroup {
+public final class WorkStealingGroup implements TaskScheduler {
```

Comment:
:question: Just so I get it right, we have no longer an Io nor CpuScheduler. The WorkStealingGroup does this right? Maybe we should give it a better name.

Reply:
Yes, that's right. Ultimately, both the IO and CPU schedulers already delegated to `WorkStealingGroup`'s `getNextTask` so this is essentially just removing indirection. Regarding naming: I feel like this is one of the few cases in the actor scheduler where the naming is spot on. Let me elaborate:

`WorkStealingGroup` is a just a collection of tasks that are available to an `ActorThreadGroup` (for us that is one group for IO and one for CPU bound). It implements `TaskScheduler` because it is in charge providing the next task that should be run. It is called `WorkStealingGroup` because it does that by potentially stealing tasks from other threads in the same group.

Does that make sense to you?

Reply:
I guess then the interface TaskScheduler is misleading, since it just provides the task but doesn't schedules it.

Reply:
Yep, maybe `TaskProvider` or something? Naming is difficult :sweat_smile: